cmake/Protobuf.cmake

```diff
@@ -11,10 +11,15 @@ macro(p4c_obtain_protobuf)
       set(SAVED_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
       set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
     endif()
-    find_package(Protobuf 3.0.0 REQUIRED)
+    find_package(Protobuf CONFIG)
```

Comment:
>Did you find any problems with the newest version of protobuf and the original version specification?

Yep, I currently have Protobuf 23.4 installed, and this doesn't seem to be compatible with `3.0.0`:

```
CMake Warning at cmake/Protobuf.cmake:14 (find_package):
  Could not find a configuration file for package "Protobuf" that is
  compatible with requested version "3.0.0".

  The following configuration files were considered but not accepted:

    /usr/lib/cmake/protobuf/protobuf-config.cmake, version: 23.4.0
    /lib/cmake/protobuf/protobuf-config.cmake, version: 23.4.0

Call Stack (most recent call first):
  CMakeLists.txt:140 (p4c_obtain_protobuf)
```

When specifying 23.0.0 instead, it also doesn't work. So far, only `23.4.0` seems to work, but will probably stop working again with `23.5` or `24`.

The version when using CMake's FindProtobuf (i.e. when not using `CONFIG`) is also different from the version when using Protobuf's own config. CMake's FindProtobuf apparently uses the classic version numbering, where `23.4` is equivalent to  `4.23.4`. Here, using `3.0.0` as the minimum version works.

Reply:
I see, that is unfortunate. I presume we don't want to have to update the version upper bound any time protobuf releases, so we will probably have to deal with unconstrained protobuf version. This might not be a big problem after all, as presumably people using `P4C_USE_PREINSTALLED_PROTOBUF=ON` do that for packaging reasons are likely to test their package :-).

Reply:
> This might not be a big problem after all, as presumably people using `P4C_USE_PREINSTALLED_PROTOBUF=ON` do that for packaging reasons are likely to test their package :-).

Yep, I also think so!